tools/depcheck/pkg/graph/graph.go

```diff
@@ -55,7 +59,99 @@ func (g *MutableDirectedGraph) AddNode(n *Node) error {
 	return nil
 }
 
+// RemoveNode deletes the given node from the graph,
+// removing all of its outbound and inbound edges as well.
+func (g *MutableDirectedGraph) RemoveNode(n *Node) {
+	delete(g.nodesByName, n.UniqueName)
+	g.DirectedGraph.RemoveNode(n)
+}
+
 func (g *MutableDirectedGraph) NodeByName(name string) (graph.Node, bool) {
 	n, exists := g.nodesByName[name]
 	return n, exists && g.DirectedGraph.Has(n)
 }
+
+// PruneOrphans recursively removes nodes with no inbound edges.
+// Receives a list of UniqueNames of nodes to ignore.
+// Returns a list of recursively pruned nodes.
+func (g *MutableDirectedGraph) PruneOrphans(originatingNodes []string) []*Node {
```

Comment:
any reason originatingNodes can't be on `g`?

Reply:
> any reason originatingNodes can't be on g?

That sounds a bit better. Will update